include/interfaces/ALE_navier_stokes.h

```diff
@@ -100,8 +217,135 @@ template <int dim, int spacedim, typename LAC>
 void ALENavierStokes<dim,spacedim,LAC>::
 set_matrix_couplings(std::vector<std::string> &couplings) const
 {
-  couplings[0] = "1,1,1; 1,1,1; 1,1,1"; // TODO: Select only not null entries
-  couplings[1] = "0,0,0; 0,0,0; 0,0,1";
+  couplings[0] = "1,1,1,1; 1,1,1,1; 1,1,1,1; 1,1,1,1"; // TODO: Select only not null entries
+  couplings[1] = "0,0,0,0; 0,0,0,0; 0,0,0,0; 0,0,0,1";
+}
+
+template <int dim, int spacedim, typename LAC>
+void
+ALENavierStokes<dim,spacedim,LAC>::
+solution_preprocessing(FEValuesCache<dim,spacedim> &fe_cache) const
+{
+  Tensor<1, dim, double> global_force;
+
+  typedef
+  FilteredIterator<typename DoFHandler<dim, spacedim>::active_cell_iterator>
+  CellFilter;
+
+  auto local_copy = [this, &global_force]
+                    (const CopyForce<dim> &data)
+  {
+    global_force += data.local_force;
+  };
+
+  auto local_assemble = [ this, &fe_cache]
+                        (const typename DoFHandler<dim, spacedim>::active_cell_iterator & cell,
+                         FEValuesCache<dim,spacedim> &scratch,
+                         CopyForce<dim> &data)
+  {
+    assemble_local_forces(cell, scratch, data, fe_cache);
```

Comment:
Questo sembra interessante…. azz!


On 16 February 2016 at 22:32:59, Alberto Sartori (notifications@github.com) wrote:

In include/interfaces/ALE_navier_stokes.h:

> +  typedef
> +  FilteredIterator<typename DoFHandler<dim, spacedim>::active_cell_iterator>
> +  CellFilter;
> +
> +  auto local_copy = [this, &global_force]
> +                    (const CopyForce<dim> &data)
> +  {
> +    global_force += data.local_force;
> +  };
> +
> +  auto local_assemble = [ this, &fe_cache]
> +                        (const typename DoFHandler<dim, spacedim>::active_cell_iterator & cell,
> +                         FEValuesCache<dim,spacedim> &scratch,
> +                         CopyForce<dim> &data)
> +  {
> +    assemble_local_forces(cell, scratch, data, fe_cache);
same here. sarebbe `(cell, fe_cache, data)

—
Reply to this email directly or view it on GitHub.